core/templates/dev/head/exploration_editor/main_tab/StateInteraction.js

```diff
@@ -119,42 +115,36 @@ oppia.controller('StateInteraction', [
     // active state is a terminal one.
     var updateDefaultTerminalStateContentIfEmpty = function() {
       // Get current state.
-      var activeStateName = editorContextService.getActiveStateName();
-      var state = explorationStatesService.getState(activeStateName);
+      var stateName = editorContextService.getActiveStateName();
 
       // Check if the content is currently empty, as expected.
-      if (state.content.length != 1 ||
-          state.content[0].value !== '' ||
-          state.content[0].type != 'text') {
+      var previousContent = explorationStatesService.getStateContentMemento(
+        stateName);
+      if (previousContent.length != 1 || previousContent[0].value !== '' ||
+          previousContent[0].type != 'text') {
         return;
       }
 
       // Update the state's content.
-      var previousContent = angular.copy(state.content);
-      state.content = [{
+      explorationStatesService.saveStateContent(stateName, [{
         type: 'text',
         value: DEFAULT_TERMINAL_STATE_CONTENT
-      }];
+      }]);
 
-      // Fire property change for editing the state's content.
-      changeListService.editStateProperty(
-        activeStateName, 'content',
-        angular.copy(state.content), previousContent);
-
-      // Save state.
-      explorationStatesService.setState(activeStateName, state);
+      // Update the state content editor view.
+      $rootScope.$broadcast('refreshStateContent');
```

Comment:
Because this can happen from outside the controller, e.g. when updating the content text for an EndExploration interaction with no content.